### PR TITLE
pkgdiff: Add a --build-system argument

### PR DESCRIPTION
**Pull request by mattst88**
I find this to be useful in cutting through a lot of the noise of diffing whole projects' sources.

### CODE REVIEW

File: pkgdiff [hunk 1]
```diff
@@ -3,6 +3,12 @@
 
 set -e -x
 
+build_system=0
+if [[ $1 == --build-system ]]; then
+	build_system=1
+	shift
+fi
```
Comment (mgorny) [inline pkgdiff]: I'm sorry but that's not consistent with my coding style. I think the following would be more future-proof:

```suggestion
mode=
while [[ ${#} -gt 0 ]]; do
	case ${1} in
		--build-system)
			mode=build-system
			;;
		*)
			break
			;;
	esac
	shift
done
```

File: pkgdiff [hunk 2]
```diff
@@ -22,4 +28,18 @@ ebuild "${1}" unpack
 ebuild "${2}" unpack
 s1=$(sed -nr 's/^declare -x S="(.*)"/\1/p' "${PORTAGE_TMPDIR}"/portage/${cat1}/${fn1%.ebuild}/temp/environment)
 s2=$(sed -nr 's/^declare -x S="(.*)"/\1/p' "${PORTAGE_TMPDIR}"/portage/${cat2}/${fn2%.ebuild}/temp/environment)
-diff --color=always --strip-trailing-cr -dupNr "${s1}" "${s2}" | ${PAGER:-less}
+
+{
+	if [[ $build_system -eq 1 ]]; then
```
Comment (mgorny) [inline pkgdiff]: And then `case` over `${mode}`.

File: pkgdiff [hunk 3]
```diff
@@ -22,4 +28,18 @@ ebuild "${1}" unpack
...
+		find "${s1}" "${s2}" -type f \
+			! \( \
+					-name 'configure.ac' -o \
+					-name 'meson.build' -o \
+					-name 'meson_options.txt' -o \
+					-name '*.cmake' -o \
+					-name 'CMakeLists.txt' \
+			\) -printf "%f\n" | diff --color=always --strip-trailing-cr -X - -dupNr "${s1}" "${s2}"
+	else
+		diff --color=always --strip-trailing-cr -dupNr "${s1}" "${s2}"
```
Comment (mgorny) [inline pkgdiff]: Perhaps you could move `diff` out of the scope, and just pass an empty list in standard mode?

**Review by mgorny — CHANGES_REQUESTED** [review]

File: pkgdiff (hunk 3 above)
Comment (mattst88) [reply pkgdiff]: Yep, looks like I don't really even need to do anything special.

**Review by mgorny — APPROVED** [review]

**Pull request closed without merging**